### PR TITLE
ci: add compliance checking via GH actions

### PR DESCRIPTION
**Pull request by nashif**
Move check_compliance script to main tree and adapt/use with GH actions.

### CODE REVIEW

**Comment by nashif** [comment]
license check can be ignored, it is about this in the comments:

```
    # <Overview of symbols defined in the file, preferably in plain English>
    (Blank line)
    # Copyright (c) 2019 ...
    # SPDX-License-Identifier: <License>
    (Blank line)
    (Kconfig definitions)
```

**Comment by nashif** [comment]
in a bad day it will look like this: https://github.com/nashif/zephyr/actions/runs/132462839

**Review by carlescufi — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

File: .github/workflows/compliance.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        pip3 install west
+
+    - name: Run Compliance Tests
+      id: compliance
+      run: |
+        export PATH=$PATH:~/.local/bin
+        export ZEPHYR_BASE=$PWD
+        git config --global user.email "you@example.com"
+        git config --global user.name "Your Name"
+        git rebase origin/master
```
Comment (stephanosio) [inline .github/workflows/compliance.yml:33]: What about pull requests to non-master branch? (e.g. backports)?

Should this use `${GITHUB_REF}` or something similar instead?

Comment (nashif) [reply .github/workflows/compliance.yml:33]: yeah, forgot about this, let me fix it.
